packages/app/src/cli/commands/app/generate/extension.ts

```diff
@@ -17,16 +15,17 @@ import {AppInterface} from '../../../models/app/app.js'
 import {load as loadApp} from '../../../models/app/loader.js'
 import generateExtensionService, {ExtensionFlavor} from '../../../services/generate/extension.js'
 import {getUIExtensionTemplates} from '../../../utilities/extensions/template-configuration.js'
-import {
-  mapExternalExtensionTypeToExtensionType,
-  mapExtensionTypesToExternalExtensionTypes,
-  mapExtensionTypeToExternalExtensionType,
-} from '../../../utilities/extensions/name-mapper.js'
+import {mapExtensionTypesToExternalExtensionTypes} from '../../../utilities/extensions/name-mapper.js'
 import metadata from '../../../metadata.js'
 import Command from '../../../utilities/app-command.js'
-import {output, path, cli, error, environment} from '@shopify/cli-kit'
+import {ensureGenerateEnvironment} from '../../../services/environment.js'
+import {fetchExtensionSpecifications} from '../../../utilities/extensions/fetch-extension-specifications.js'
+import {allFunctionSpecifications} from '../../../models/extensions/specifications.js'
+import {FunctionSpec} from '../../../models/extensions/functions.js'
+import {output, path, cli, error, environment, session} from '@shopify/cli-kit'
 import {Flags} from '@oclif/core'
 import {PackageManager} from '@shopify/cli-kit/node/node-package-manager'
+import {RemoteSpecification} from '@shopify/cli-kit/src/api/graphql/extension_specifications.js'
 
 export default class AppScaffoldExtension extends Command {
```

Comment:
I'd recommend looking at commands as views in the MVC and having them as lean as possible. All they should do is declare themselves (flags, arguments, description), and delegate the business logic to a service function. Looking at the implementation of the function here, it's starting to have too much logic and it might be a good time to extract the functionality. Besides being able to write tests that don't have to instantiate an Oclif command class, we can also reuse the logic in other contexts.

Reply:
You are totally right. In this case, all the logic was already in the command file, and to avoid a bigger PR and refactor, I just updated what I needed here.
But I have a list of "things to clean up" after the extensions work is done, including this file and the `loader.ts` for instance.

If it's ok with you, i'll do this refactor in a different PR with proper tests for the service (I feel it will be a BIG PR)